website/blog/2024-03-18-deep-learning-part-2/index.md

```diff
@@ -0,0 +1,230 @@
+---
+title: Fine Tuning LLMs with MLflow: Deep Learning with MLflow (Part 2)
+description: Highlighting the recent improvements in MLflow for Deep Learning workflows
+slug: deep-learning-part-1
+authors: [abe-omorogbe, avinash-sooriyarachchi, puneet-jain]
+tags: [Deep Learning]
+thumbnail: img/blog/dl-chart-grouping.gif
+---
+
+![](deep-learning-pt2-min.jpg)
+
+In the realm of deep learning, finetuning of pre-trained Large Language Models (LLMs) on private datasets stands as a critical application. This practice is not only common but also essential for developing specialized models, particularly for tasks like custom Named Entity Recognition (NER). In such scenarios, tools like MLflow are invaluable. They ensure that every aspect of the training process - metrics, parameters, and artifacts - are reproducibly tracked, logged, analyzed, compared, and shared. In this tutorial we are going to be using [MLflow 2.11](https://mlflow.org/releases/2.11.0) and [the new MLflow Deep Learning features](https://mlflow.org/blog/deep-learning-part-1) to track all the important aspects of this project.
+
+
+# Use Case: Fine Tuning Llama2-7B for NER on Restaurant data
+
+Let's consider the use case of fine-tuning a Llama2-7B model for use as a tool for extracting instructive contexts that are embedded within bodies of text. This is a specific example of how Parameter-Efficient Fine-Tuning (PEFT) can be applied for real-world tasks. The goal here is to  extract structured information based on instructions, which goes beyond simple Named Entity Recognition (NER).
+
+Imagine a scenario where the instruction is: `The Punter is an expensive Chinese coffee shop located near Café Sicilia.` The model, after fine-tuning, would then process the unstructured text of the sentence and identify the entities, presenting them in a structured format like a list. 
+
+Example of extracting information from `The Punter is an expensive Chinese coffee shop located near Café Sicilia.`: 
+```python
+Business name: The Punter
+Cuisine: Chinese
+descriptor: expensive
+location: Café Sicilia
+```
+
+The Llama2-7B model's size is strategically chosen. It has enough parameters (complexity) to handle intricate tasks like understanding instructions within text, but remains compact enough for fine-tuning on even a modest cluster of less powerful GPUs like Nvidia T4s, thanks to the PEFT methodology. It can even be fine-tuned on a single high-end A100 GPU with PEFT.
```

Comment:
```suggestion
The Llama2-7B model's size is strategically chosen. It has enough parameters (complexity) to handle intricate tasks like understanding instructions within text, but remains compact enough for fine-tuning on even a modest cluster of less powerful GPUs (such as Nvidia T4s), thanks to the PEFT methodology. Alternatively, it can be fine-tuned on a single high-end A100 GPU by using PEFT (as opposed to a non-PEFT fine tuning operation, which would require significantly more resources).
```